src/Features/Core/Portable/Completion/CompletionRules.cs

```diff
@@ -20,20 +21,22 @@ internal class CompletionRules
         private readonly object _gate = new object();
         private readonly AbstractCompletionService _completionService;
         private readonly Dictionary<string, PatternMatcher> _patternMatcherMap = new Dictionary<string, PatternMatcher>();
+        private readonly Dictionary<string, PatternMatcher> _enUSCulturePatternMatcherMap = new Dictionary<string, PatternMatcher>();
+        private readonly CultureInfo EnUSCultureInfo = new CultureInfo("en-US");
```

Comment:
This is instance, when it doesn't change for lifetime of the appdomain. Make it static.